dns_cache.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"context"
+	"net"
+	"time"
+
+	"github.com/rs/dnscache"
+)
+
+// How often should we check for successful updates to cached entries
+const dnsCacheRefreshInterval = 5 * time.Minute
+
+// Local DNS cache because in this world things are ephemeral
+type cachedDNS struct {
+	resolver  *dnscache.Resolver
+	refresher *time.Ticker
+}
+
+func newCachedDNS(refreshInterval time.Duration) *cachedDNS {
```

Comment:
Normally we would install a local resolver with its own caching in place but it wouldn't be needed in any case with this, iirc?

Reply:
yeah, if we want to drop this and just have the OS take care of it that's also totally fine. Leaving it for now though